packages/js/src/Modules/Verto/BaseSession.ts

```diff
@@ -206,6 +201,22 @@ export default abstract class BaseSession {
    *   // Got an error...
    * })
    * ```
+   *
+   * ## Events
+   * |   |   |
+   * |---|---|
+   * | `telnyx.ready` | The client is authenticated and available to use |
+   * | `telnyx.error` | An error occurred at the session level |
+   * | `telnyx.notification` | An update to the call or session |
+   * | `telnyx.socket.open` | The WebSocket connection has been made |
+   * | `telnyx.socket.close` | The WebSocket connection is set to close |
+   * | `telnyx.socket.error` | An error occurred at the WebSocket level |
+   * | `telnyx.socket.message` | The client has received a message through WebSockets |
+   *
+   * @param eventName Event name.
+   * @param callback Function to call when the event comes.
+   *
+   * @return The client object itself.
    */
   on(eventName: string, callback: Function) {
```

Comment:
We could type eventName as an enum. I guess we would still have to document what each means though.

Having it as an Enum here will provide a nicer developer experience, as you get autocomplete when providing the string or can use `Enum.Event` kind of syntax.

Reply:
I don't think this is possible with string `Enum` values, but will investigate